web3/utils/formatters.py

```diff
@@ -67,7 +67,10 @@ def apply_formatter_if(condition, formatter, value):
 def apply_formatters_to_dict(formatters, value):
     for key, item in value.items():
         if key in formatters:
-            yield key, formatters[key](item)
+            try:
+                yield key, formatters[key](item)
+            except (TypeError, ValueError) as exc:
+                raise type(exc)("Could not format value %r as field %r" % (item, key)) from exc
```

Comment:
I think this should be changed to include the original exception message (it doesn't appear it currently does)

```python
raise type(exc)("Could not format value %r as field %r: %r" % (item, key, str(exc))) from exc
```

Reply:
Since it is a `raise from` the old trace and message all still show up though. If you missed the `raise from`, do you think this is easier to read?

```
except (TypeError, ValueError) as old_exception:
    new_exception = type(exc)("Could not format value %r as field %r: %r" % (item, key, str(exc)))
    raise new_exception from old_exception
```

Reply:
I didn't miss it, I just wasn't aware that it showed both messages.  Current format is good with me.